integration/spark/src/main/scala/org/apache/spark/sql/CarbonDictionaryDecoder.scala

```diff
@@ -193,15 +194,21 @@ case class CarbonDictionaryDecoder(
               }
               flag
             }
+
             override final def next(): InternalRow = {
               val startTime = System.currentTimeMillis()
               val row: InternalRow = iter.next()
               val data = row.toSeq(dataTypes).toArray
               dictIndex.foreach { index =>
                 if (data(index) != null) {
-                  data(index) = DataTypeUtil.getDataBasedOnDataType(dicts(index)
-                    .getDictionaryValueForKey(data(index).asInstanceOf[Int]),
-                    getDictionaryColumnIds(index)._3)
+                  try {
+                    data(index) = DataTypeUtil.getDataBasedOnDataType(dicts(index)
```

Comment:
Please just add  `!carbonDimension.isComplex()` in line number 63 and 132. That should solve the problem, remove all other unnecessary excepion handlings